# Setup environment variabel

In [5]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [7]:
sns.set_style("whitegrid")
sns.despine()
plt.style.use("seaborn-whitegrid")
plt.rc("figure", autolayout=True)
plt.rc("axes", labelweight="bold", labelsize="large", titleweight="bold", titlesize=14, titlepad=10)
from scipy import stats

import regex as re

from sklearn.preprocessing import LabelEncoder
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split
from sklearn.decomposition import PCA

import xgboost
from xgboost import XGBRegressor
from sklearn.ensemble import RandomForestRegressor

import optuna

from sklearn.metrics import mean_squared_error
from sklearn.metrics import r2_score

seed=42

C:\Users\Hp\AppData\Local\Temp\ipykernel_13760\1200699343.py:3: MatplotlibDeprecationWarning: The seaborn styles shipped by Matplotlib are deprecated since 3.6, as they no longer correspond to the styles shipped by seaborn. However, they will remain available as 'seaborn-v0_8-<style>'. Alternatively, directly use the seaborn API instead.
  plt.style.use("seaborn-whitegrid")
e:\A6\Analisis dan p.data\Project_TA\env\lib\site-packages\tqdm\auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


<Figure size 640x480 with 0 Axes>

**General infos**:
- The majority of laptops are from Dell, lenovo and HP.
- The majority of laptops are Notebooks, which make 55.84% of the total laptops
- Most laptops have 15.6 inches, which make 51.08% of the total laptops

**Price**:
- There are laptops with prices over 3000:
- Most of them are Gaming PCs or Workstations with intel CPU.
- There is also one Notebook with a price close to 5000 euros and Gaming laptops, with a price close to 5500 euros and 6000 euros
- The most expensive notebook is the Lenovo Thinkpad P51, with an Intel XEON CPU and Nvidia Quadro GPU!

**Brand** :
- Laptops with Intel CPUs are more expensive.
- Laptops with an AMD CPUs also hahave and AMD GPUs
- Laptops with Nvidia GPUs are more expensive.

**RAM**:
- According to the slope of the linear regression between price and RAM, every GB of RAM added on the PC adds roughly 107$ to the laptop value.
- Most laptops have 8 GB RAM, which make 47.54% of the total laptops

**GPU**:
- The 2 most common GPUs are integrated Intel GPUs HD Graphics 620 and 520, while the third one is the Nvidia GTX1050.
**CPU**:
- All the TOP 15 most common CPUs are from Intel.
- The most common CPU is the Intel i5 7200U, second is i7 7700HQ and third is i7 7500U.
- Out of the 15 CPUs, 10 are series 'U' (low voltage) , 3 are series 'HQ' (high performance) 10 and 2 are Celerons (low-end).
- Most laptops have 2.5 GHz CPU, which make 22.5% of the total laptops

**Hard drives**:
- Most PCs have 256 GB of storage, which is for the most part SSD. Moreover, for storage of 1 TB or higher, most of them are HDD.
- Most of second hard drive storages are 1 TB HDD disks

**Correlation Matrix**:
- RAM has a high positive correlation with price (+0.75): more expensive laptops tends to have a higher pric
- CPU Freq has a quite moderate positive correlation with the price (+0.45)
- Inches and Weight have a high positive correlation (+0.82) since laptops with bigger screens tend to be heavier.

In [3]:
data = pd.read_csv("laptop_price.csv",encoding='latin1')
data.head(10)

laptop_ID Company          Product   TypeName  Inches  \
0          1   Apple      MacBook Pro  Ultrabook    13.3   
1          2   Apple      Macbook Air  Ultrabook    13.3   
2          3      HP           250 G6   Notebook    15.6   
3          4   Apple      MacBook Pro  Ultrabook    15.4   
4          5   Apple      MacBook Pro  Ultrabook    13.3   
5          6    Acer         Aspire 3   Notebook    15.6   
6          7   Apple      MacBook Pro  Ultrabook    15.4   
7          8   Apple      Macbook Air  Ultrabook    13.3   
8          9    Asus  ZenBook UX430UN  Ultrabook    14.0   
9         10    Acer          Swift 3  Ultrabook    14.0   

                     ScreenResolution                         Cpu   Ram  \
0  IPS Panel Retina Display 2560x1600        Intel Core i5 2.3GHz   8GB   
1                            1440x900        Intel Core i5 1.8GHz   8GB   
2                   Full HD 1920x1080  Intel Core i5 7200U 2.5GHz   8GB   
3  IPS Panel Retina Display 2880x1800        Intel Core i7 2.7GHz  16GB   
4  IPS Panel Retina Display 2560x1600        Intel Core i5 3.1GHz   8GB   
5                            1366x768     AMD A9-Series 9420 3GHz   4GB   
6  IPS Panel Retina Display 2880x1800        Intel Core i7 2.2GHz  16GB   
7                            1440x900        Intel Core i5 1.8GHz   8GB   
8                   Full HD 1920x1080  Intel Core i7 8550U 1.8GHz  16GB   
9         IPS Panel Full HD 1920x1080  Intel Core i5 8250U 1.6GHz   8GB   

                Memory                           Gpu       OpSys  Weight  \
0            128GB SSD  Intel Iris Plus Graphics 640       macOS  1.37kg   
1  128GB Flash Storage        Intel HD Graphics 6000       macOS  1.34kg   
2            256GB SSD         Intel HD Graphics 620       No OS  1.86kg   
3            512GB SSD            AMD Radeon Pro 455       macOS  1.83kg   
4            256GB SSD  Intel Iris Plus Graphics 650       macOS  1.37kg   
5            500GB HDD                 AMD Radeon R5  Windows 10   2.1kg   
6  256GB Flash Storage       Intel Iris Pro Graphics    Mac OS X  2.04kg   
7  256GB Flash Storage        Intel HD Graphics 6000       macOS  1.34kg   
8            512GB SSD          Nvidia GeForce MX150  Windows 10   1.3kg   
9            256GB SSD        Intel UHD Graphics 620  Windows 10   1.6kg   

   Price_euros  
0      1339.69  
1       898.94  
2       575.00  
3      2537.45  
4      1803.60  
5       400.00  
6      2139.97  
7      1158.70  
8      1495.00  
9       770.00

In [8]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1303 entries, 0 to 1302
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   laptop_ID         1303 non-null   int64  
 1   Company           1303 non-null   object 
 2   Product           1303 non-null   object 
 3   TypeName          1303 non-null   object 
 4   Inches            1303 non-null   float64
 5   ScreenResolution  1303 non-null   object 
 6   Cpu               1303 non-null   object 
 7   Ram               1303 non-null   object 
 8   Memory            1303 non-null   object 
 9   Gpu               1303 non-null   object 
 10  OpSys             1303 non-null   object 
 11  Weight            1303 non-null   object 
 12  Price_euros       1303 non-null   float64
dtypes: float64(2), int64(1), object(10)
memory usage: 132.5+ KB


- There are just 2 numerical columns: 'inches' and 'price euros'.
- we must change the categorical to numerical number

In [12]:
data.describe()

laptop_ID       Inches  Price_euros
count  1303.000000  1303.000000  1303.000000
mean    660.155794    15.017191  1123.686992
std     381.172104     1.426304   699.009043
min       1.000000    10.100000   174.000000
25%     331.500000    14.000000   599.000000
50%     659.000000    15.600000   977.000000
75%     990.500000    15.600000  1487.880000
max    1320.000000    18.400000  6099.000000

- LaptopID: We can se that also laptopID is included as a numerical features, with a minimum value of 1 and max of 1320, different from the index (0-1303). This will be investigated later.

- Inches: the nimimum inches is 10,1, while the max is 18.4 and the mean is around 15 inches

- Price_euros:miimuim price is 174 euros, the max is 6099 and the mean is 1123, which is kinda high for a laptop